src/st/no0/4E2E0.c

```diff
@@ -1,6 +1,92 @@
 // SPDX-License-Identifier: AGPL-3.0-or-later
 #include "no0.h"
 
-INCLUDE_ASM("st/no0/nonmatchings/4E2E0", func_us_801CE2E0);
+static u8 anim_unk[] = {
+    0x01, 0x01, 0x01, 0x02, 0x01, 0x03, 0x01, 0x04, 0x01, 0x05, 0x01, 0x06,
+    0x01, 0x07, 0x01, 0x08, 0x01, 0x09, 0x01, 0x0A, 0x01, 0x0B, 0x01, 0x0C,
+    0x01, 0x0D, 0x01, 0x0E, 0x01, 0x0F, 0x01, 0x10, 0x00, 0x00, 0x00, 0x00};
 
-INCLUDE_RODATA("st/no0/nonmatchings/4E2E0", D_us_801C14A8);
+// Does not appear in the entity list and appears to be unused
+void func_us_801CE2E0(Entity* self) {
+    s16 yOffsets[8] = {-2, -1, -1, 0, 0, 1, 1, 2};
+    Entity* player;
+    s32 collision;
+    s16 offset;
+    u16 flags;
+
+    if (self->step != 0) {
+        UnkAnimFunc(anim_unk, self, 5);
+        self->rotZ += 0x80;
+        player = &PLAYER;
+        if (self->ext.et_801CE2E0.unk88 & 64) {
+            self->velocityY = FIX(-0.5);
+        } else {
+            self->velocityY = FIX(0.5);
+        }
+
+        self->velocityX = 0;
+        self->ext.et_801CE2E0.unk88++;
+        MoveEntity();
+        if (self->ext.et_801CE2E0.unk7C & 4) {
+            offset = (self->posY.i.hi + g_Tilemap.scrollY.i.hi) -
+                     self->ext.et_801CE2E0.posY;
+            if (offset > 0 || (offset < 0 && (g_Player.pl_vram_flag ^ 2) & 2)) {
+                player->posY.i.hi += offset;
+            }
+        }
+        flags = 15;
+        if (self->ext.et_801CE2E0.unk8A != 0) {
+            self->ext.et_801CE2E0.unk8A--;
+            flags = 11;
+        }
+
+        if (self->ext.et_801CE2E0.unk80 & 15 &&
```

Comment:
Probably best to use hex notation for bitmasks - so using `& 0xF` and `& 0xF0` in this case, and for the others in this function.